server/src/main/java/org/elasticsearch/index/query/InterceptedQueryBuilderWrapper.java

```diff
@@ -63,7 +60,7 @@ public Query toQuery(SearchExecutionContext context) throws IOException {
 
     @Override
     public QueryBuilder queryName(String queryName) {
-        return queryBuilder.queryName(queryName);
+        return new InterceptedQueryBuilderWrapper(queryBuilder.queryName(queryName));
```

Comment:
Do we need a new wrapper? Can't we just return `this`?

```suggestion
        queryBuilder.queryName(queryName);
        return this;
```